streams/src/main/java/org/apache/kafka/streams/state/internals/KeyValueToTimestampedKeyValueByteStoreAdapter.java

```diff
@@ -127,14 +131,31 @@ public <R> QueryResult<R> query(
         final QueryConfig config) {
 
 
+
         final long start = config.isCollectExecutionInfo() ? System.nanoTime() : -1L;
-        final QueryResult<R> result = store.query(query, positionBound, config);
+        QueryResult<R> result = store.query(query, positionBound, config);
+        final Position position = result.getPosition();
+
+        if (result.isSuccess()) {
+            if (result.getResult() instanceof byte[]) {
+                final byte[] plainValue = (byte[]) result.getResult();
+                final byte[] valueWithTimestamp = convertToTimestampedFormat(plainValue);
+                result = (QueryResult<R>) QueryResult.forResult(valueWithTimestamp);
+            }
+            if (result.getResult() instanceof RocksDbIterator) {
+                final WrappedRocksDbIterator wrappedRocksDBRangeIterator = new WrappedRocksDbIterator((RocksDbIterator) result.getResult());
+                result = (QueryResult<R>) QueryResult.forResult(wrappedRocksDBRangeIterator);
+            }
```

Comment:
We should chain this with an `else` that would throw an exception (we should never hit the `else` and it would be a bug -- throwing an exception would surface the bug clearly).

Reply:
ok